packages/libraries/cli/src/helpers/target-input.ts

```diff
@@ -0,0 +1,52 @@
+import { z } from 'zod';
+import * as GraphQLSchema from '../gql/graphql';
+
+type ParseError = {
+  type: 'error';
+};
+
+type ParseOk = {
+  type: 'ok';
+  data: GraphQLSchema.TargetReferenceInput;
+};
+
+const UUIDModel = z.string().uuid();
+
+/**
+ * Parse a target slug into its parts. Returns an error if slug is invalid
+ */
+export function parse(str: string): ParseError | ParseOk {
+  const uuidResult = UUIDModel.safeParse(str);
+
+  if (uuidResult.success) {
+    return {
+      type: 'ok',
+      data: {
+        byId: uuidResult.data,
+      },
+    };
+  }
+
+  const parts = str.split('/');
+
+  const organizationSlug = parts.at(0);
+  const projectSlug = parts.at(1);
+  const targetSlug = parts.at(2);
```

Comment:
i think we should validate `parts.length === 3` to make sure there isnt confusion with additional `/stuff`

Reply:
116575b28 🫡